cmd/symbols/shared/setup.go

```
@@ -127,10 +131,17 @@ func setupRockskip(observationCtx *observation.Context, config rockskipConfig, g
 	observationCtx = observation.ContextWithLogger(observationCtx.Logger.Scoped("rockskip"), observationCtx)
 
 	codeintelDB := mustInitializeCodeIntelDB(observationCtx)
-	createParser := func() (ctags.Parser, error) {
-		return symbolsParser.SpawnCtags(log.Scoped("parser"), config.Ctags, ctags_config.UniversalCtags)
+
+	logger := log.Scoped("parser")
+	parserFactory := func(source ctags_config.ParserType) (ctags.Parser, error) {
+		return symbolparser.SpawnCtags(logger, config.Ctags, source)
+	}
+	symbolParserPool, err := symbolparser.NewParserPool(observationCtx, "src_rockskip_service", parserFactory, config.NumCtagsProcesses, parserTypesForDeployment())
+	if err != nil {
+		logger.Fatal("failed to create symbol parser pool", log.Error(err))
```

shouldn't this return an error instead of using logger.Fatal?

---

Good catch